### PR TITLE
[issue-552] Support SpecificRecordBase in DeserializerFromSchemaRegistry

### LINKED ISSUE #552: Support SpecificRecordBase in DeserializerFromSchemaRegistry
**Problem description**
In `DeserializerFromSchemaRegistry`, currently we create an `AvroGenericDeserializer` when type class is `GenericRecord` or its subclasses, which will ignore the circumstance where type class is `SpecificRecordBase` since it is also a subclass of `GenericRecord`.

**Problem location**
`DeserializerFromSchemaRegistry`

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by fyang86**
Signed-off-by: Fan, Yang <fan.yang5@emc.com>

**Change log description**
Add support for avro format `SpecificRecordBase` in `DeserializerFromSchemaRegistry`.

**Purpose of the change**
Fix #552 

**What the code does**
Change if order in `DeserializerFromSchemaRegistry`.
Update unit test `FlinkPravegaSchemaRegistryReaderTestITCase`

**How to verify it**
`./gradlew clean build` passed

### CODE REVIEW

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java [hunk 1]
```diff
@@ -51,19 +53,62 @@ public MyTest(String a) {
         }
     }
 
+    @SuppressWarnings("checkstyle:ConstantName")
```
Comment (crazyzhou) [inline src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java:56]: We don't need to keep the constant name the same with the avro-generated class, a constant named `SCHEMA` would be enough. the class name can also be simplified to `User` as well.
Another path is that we keep this class real avro generated, we need to have a `.avsc` file in the `resources` folder and let avro generate the class. See 

https://github.com/apache/flink/blob/master/flink-formats/flink-avro-glue-schema-registry/src/test/java/resources/avro/user.avsc
https://github.com/apache/flink/blob/master/flink-formats/flink-avro-glue-schema-registry/src/test/java/org/apache/flink/formats/avro/glue/schema/registry/User.java

Both approaches are good to me.

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java [hunk 2]
```diff
@@ -51,19 +53,62 @@ public MyTest(String a) {
...
             new SchemaRegistryUtils(SETUP_UTILS, SchemaRegistryUtils.DEFAULT_PORT);
 
-    private static final Schema SCHEMA = SchemaBuilder
-            .record("MyTest")
-            .fields()
-            .name("a")
-            .type(Schema.create(Schema.Type.STRING))
-            .noDefault()
-            .endRecord();
+    private static final Schema SCHEMA = new Schema.Parser().parse(
```
Comment (crazyzhou) [inline src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java:108]: Can we use `SpecificRecordTestClass::SCHEMA` here? Looks like they are the same

**Review by crazyzhou — CHANGES_REQUESTED** [review]

File: src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java (hunk 1 above)
Comment (fyang86) [reply src/test/java/io/pravega/connectors/flink/FlinkPravegaSchemaRegistryReaderTestITCase.java:56]: agreed, will fix this

**Review by crazyzhou — APPROVED** [review]
LGTM

**Pull request merged**